### PR TITLE
feat: use cache option

### LINKED ISSUE #5: Suggestion: if statement for caching
Adding cached-uv sped up my pypy jobs (yay!), but slowed down all the other jobs, as it takes 2-10 seconds to restore the cache, and uv's just as fast without it if it doesn't have to build anything (so slower if you include the cache restore time, its almost always double the time). PyPy (and probably Python 3.13, forgot to check) need to build, so I'd like to have an `with: use-cache:` field that I could put a template into that would be true if I need the cache, and false if not (the cache is ~70MB per job, and I have a lot of jobs, so it would be nice to just do the important bits).

I can do this with something like this:

```yaml
    - name: Setup uv (uncached)
      uses: yezz123/setup-uv@v4
      if: "!startsWith(matrix.python-version, 'pypy')"

    - name: Setup uv (cached)
      uses: hynek/setup-cached-uv@v1
      if: "startsWith(matrix.python-version, 'pypy')"
      with:
        cache-suffix: -${{ matrix.python-version }}
```

but it's not as pretty to require two separate actions, IMO.

(FWIW, caching doesn't seem to be worth it for pixi or uv unless they have to build (which pixi never does...)

### PR DESCRIPTION
**Pull request by henryiii**
Close  #5. See https://github.com/tomerfiliba/plumbum/pull/684 for an example of using it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ This could, for example, be the name of the job, or the current week number to b
       - run: echo WEEK=$(date +%V) >>$GITHUB_ENV
         shell: bash
 
-      - uses: ./action
```
Comment (hynek) [inline README.md:48]: Is this on purpose?

File: action.yml [hunk 2]
```diff
@@ -17,6 +17,11 @@ inputs:
     required: false
     default: ""
 
+  use-cache:
```
Comment (hynek) [inline action.yml]: Can we call it `use-cache-if`?

File: README.md [hunk 3]
```diff
@@ -60,10 +60,25 @@ Internally, the GitHub Actions function `hashFiles` is used to hash the passed p
 
 Using this with a fully pinned `requirements.txt` file is the most efficient use of this action because it automatically invalidates the cache.
 
+#### `use-cache`
```
Comment (hynek) [inline README.md]:
```suggestion

#### `use-cache`
```

File: README.md [hunk 4]
```diff
@@ -60,10 +60,25 @@ Internally, the GitHub Actions function `hashFiles` is used to hash the passed p
...
+
+This defaults to `true`, but can be used to disable the cache, since GitHub's
```
Comment (hynek) [inline README.md]: Could you switch these to one sentence per line please?

**Review by hynek — CHANGES_REQUESTED** [review]
I’ve added some contents but as said before I think the idea is sound!

One think tho: I’m in the middle of Pantanal and am emphatically afk (brought no real computer) so my next opportunity to release is by the end of June. I could add you as a maintainer if you want one earlier tho.

oh wait and is there a changelog? Phone reviews are ughhh

**Comment by henryiii** [comment]
Haha, no rush at all (if you can't tell!). :) End of month is fine. It can be worked around, it's just a nice quality of life improvement. I'll try to address the suggestions above in the next couple of days.

**Comment by henryiii** [comment]
Ah, totally forgot about this. To be fair, I've been out with conjunctivitis for the last week+. I'll address these.

File: action.yml (hunk 2 above)
Comment (henryiii) [reply action.yml]: We can call it supercalifragilisticexpialidocious if you want, it's up to you. :D

File: README.md (hunk 1 above)
Comment (henryiii) [reply README.md:48]: The other example shows what a user would do. This is only valid for this one action, other repos won't use `./action`. Plus we went from two examples to three which made this one stick out as the only one that shows how to use the action from the action repo.

**Comment by henryiii** [comment]
I've addressed everything but the `./action` comment.

**Review by hynek — APPROVED** [review]
lovely, thank you!

**Pull request merged**